test/e2e/storage/subpath.go

```diff
@@ -307,6 +308,20 @@ var _ = utils.SIGDescribe("Subpath", func() {
 
 				testPodContainerRestart(f, pod, filePathInVolume, filePathInSubpath)
 			})
+
+			It("should unmount if pod is gracefully deleted while kubelet is down [Disruptive][Slow]", func() {
+				if curVolType == "hostPath" || curVolType == "hostPathSymlink" {
```

Comment:
There is one side issue here that hostpath volume plugin doesn't support reconstruction.  We probably need to figure out how to implement support for it otherwise the subpath bind mounts are not going to get cleaned up for hostpath volumes (although I still question why do you need to use subpath with hostpath volumes?).

Reply:
regardless of reconstruction after newest bit of refactoring - `Teardown` should be called for all volume types that are not in use, including hostpaths. Shouldn't that cleanup subpaths too?

Reply:
Yes hostpath subpath is cleaned up in normal cases.  But in the reconstruction case, it is not.  And this PR won't fix that

Reply:
I think graceful pod delete should still work for hostpath.  Only forceful delete will fail, so this check can be removed.